core/src/main/java/org/ehcache/core/EhcacheWithLoaderWriter.java

```diff
@@ -405,17 +403,11 @@ protected void doRemoveAll(final Set<? extends K> keys) throws BulkCacheWritingE
     return store.computeIfAbsent(key, mappingFunction);
   }
 
-  /**
-   * {@inheritDoc}
-   */
   @Override
-  public boolean remove(final K key, final V value) throws CacheWritingException {
-    conditionalRemoveObserver.begin();
-    statusTransitioner.checkAvailable();
-    checkNonNull(key, value);
-    final AtomicBoolean hit = new AtomicBoolean();
-    final AtomicBoolean removed = new AtomicBoolean();
-    final BiFunction<K, V, V> remappingFunction = memoize((k, inCache) -> {
+  protected Store.RemoveStatus doRemove(K key, V value) throws StoreAccessException {
+    AtomicBoolean hit = new AtomicBoolean();
```

Comment:
More `boolean[1]` possibilities (I've probably missed some others too).

Reply:
I did it for all `EhcacheLoaderWriter`. There are some opportunities in `Ehcache` as well. Two questions: Are we sure it's better? And are we sure the functions can never execute in parallel? (we could imagine a store doing a getAll in parallel)